Common types used across [Platform]MouseEvent.h should be declared in a standalone header

aprotyas opened the pull request:
#### 0bfd0b0c2c3be61710f849f4d63f510f29b0a9b1
<pre>
Common types used across [Platform]MouseEvent.h should be declared in a standalone header
<a href="https://bugs.webkit.org/show_bug.cgi?id=287059">https://bugs.webkit.org/show_bug.cgi?id=287059</a>
<a href="https://rdar.apple.com/144204277">rdar://144204277</a>

Reviewed by Wenson Hsieh and Aditya Keerthi.

There are some common types/constants used across [Platform]MouseEvent.h
that often need to be referenced throughout the codebase in other
projects (WebKit/WKTR/TestWebKitAPI/etc.), and having these types be
defined across the event headers makes for quite heavy include costs.
For example, pulling in WebCore/MouseEvent.h from DumpRenderTree also
necessitates the inclusion of WebCore/Node.h.

This patch splits out some of these common types/constants into a
separate header, namely MouseEventTypes.h. It also introduces a helper
function WebCore::buttonFromShort(), that will be used from some test
harness code in a forthcoming patch.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEventTypes.h: Added.
(WebCore::buttonFromShort):
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/PointerEvent.cpp:
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/WheelEvent.cpp:
* Source/WebCore/dom/ios/MouseEventIOS.cpp:
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLSelectElement.cpp:
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/PointerCaptureController.cpp:
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/platform/PlatformMouseEvent.h:
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/WebEventConversion.cpp:
* Source/WebKit/Shared/WebMouseEvent.cpp:
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:

Canonical link: <a href="https://commits.webkit.org/289859@main">https://commits.webkit.org/289859@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/70cfffa143520b86a7e34e7d29652ba9f7c3f2c3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88231 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/7748 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42643 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93184 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38981 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/90282 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8132 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15929 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68064 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25785 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91233 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6211 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79796 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48433 "Found 1 new API test failure: /TestJSC:/jsc/options (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5983 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/34204 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38089 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76365 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35089 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95030 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15401 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11288 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76929 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15657 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75652 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76174 "Found 1 new API test failure: /TestJSC:/jsc/options (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18736 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20563 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18974 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8428 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15419 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/20718 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15161 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18607 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16943 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/dom/MouseEventTypes.h:
```diff
@@ -0,0 +1,56 @@
+namespace WebCore {
+
+const double ForceAtClick = 1;
+const double ForceAtForceClick = 2;
```
`constexpr`?

aprotyas replied:
True, good catch.

aprotyas replied:
Fixed

whsieh reviewed (approved):
r=mews

In all the places that already include `MouseEvent.h`, I think you can omit the new header since it's going to be included transitively. Might make the patch somewhat smaller.

aprotyas committed:
Common types used across [Platform]MouseEvent.h should be declared in a standalone header
https://bugs.webkit.org/show_bug.cgi?id=287059
rdar://144204277

Reviewed by Wenson Hsieh and Aditya Keerthi.

There are some common types/constants used across [Platform]MouseEvent.h
that often need to be referenced throughout the codebase in other
projects (WebKit/WKTR/TestWebKitAPI/etc.), and having these types be
defined across the event headers makes for quite heavy include costs.
For example, pulling in WebCore/MouseEvent.h from DumpRenderTree also
necessitates the inclusion of WebCore/Node.h.

This patch splits out some of these common types/constants into a
separate header, namely MouseEventTypes.h. It also introduces a helper
function WebCore::buttonFromShort(), that will be used from some test
harness code in a forthcoming patch.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/model-element/HTMLModelElement.cpp:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/MouseEvent.h:
* Source/WebCore/dom/MouseEventTypes.h: Added.
(WebCore::buttonFromShort):
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/PointerEvent.cpp:
* Source/WebCore/dom/SimulatedClick.cpp:
* Source/WebCore/dom/WheelEvent.cpp:
* Source/WebCore/dom/ios/MouseEventIOS.cpp:
* Source/WebCore/dom/ios/PointerEventIOS.cpp:
* Source/WebCore/html/HTMLAnchorElement.cpp:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLSelectElement.cpp:
* Source/WebCore/html/shadow/TextControlInnerElements.cpp:
* Source/WebCore/page/AutoscrollController.cpp:
* Source/WebCore/page/PointerCaptureController.cpp:
* Source/WebCore/page/PointerCaptureController.h:
* Source/WebCore/page/ios/EventHandlerIOS.mm:
* Source/WebCore/page/mac/ServicesOverlayController.mm:
* Source/WebCore/platform/PlatformMouseEvent.h:
* Source/WebCore/platform/ios/PlatformEventFactoryIOS.mm:
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/WebEventConversion.cpp:
* Source/WebKit/Shared/WebMouseEvent.cpp:
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
* Source/WebKitLegacy/ios/WebView/WebPDFViewPlaceholder.mm:
* Source/WebKitLegacy/mac/WebView/WebFrame.mm:
* Source/WebKitLegacy/mac/WebView/WebPDFView.mm:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:

Canonical link: https://commits.webkit.org/289859@main
